src/persistence/surreal/keysets.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::BTreeMap;
+// ----- standard library imports
+use std::collections::HashMap;
+// ----- extra library imports
+use anyhow::Result as AnyResult;
+use async_trait::async_trait;
+use cdk::nuts::nut00 as cdk00;
+use cdk::nuts::nut01 as cdk01;
+use cdk::nuts::nut02 as cdk02;
+use surrealdb::Result as SurrealResult;
+use surrealdb::{engine::any::Any, Surreal};
+use uuid::Uuid;
+// ----- local modules
+// ----- local imports
+use crate::credit::keys as creditkeys;
+use crate::keys;
+use crate::persistence::surreal::ConnectionConfig;
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+struct DBQuoteKeys {
+    qid: surrealdb::Uuid,
+    info: cdk::mint::MintKeySetInfo,
+    // unpacking MintKeySet because surrealdb doesn't support BTreeMap<K,V> where K is not a String
+    unit: cdk00::CurrencyUnit,
+    keys: HashMap<String, cdk01::MintKeyPair>,
+}
+
+fn keysentry2dbquoteskeys(qid: Uuid, ke: keys::KeysetEntry) -> DBQuoteKeys {
+    let (info, keyset) = ke;
+    let cdk02::MintKeySet { unit, mut keys, .. } = keyset;
+    let mut serialized_keys = HashMap::new();
+    while let Some((amount, keypair)) = keys.pop_last() {
+        serialized_keys.insert(amount.to_string(), keypair);
+    }
+    DBQuoteKeys {
+        qid,
+        info,
+        unit,
+        keys: serialized_keys,
+    }
+}
+
+fn dbquoteskeys2keysentry(dbqk: DBQuoteKeys) -> (Uuid, keys::KeysetEntry) {
+    let DBQuoteKeys {
+        qid,
+        info,
+        unit,
+        keys,
+    } = dbqk;
+    let mut keysmap: BTreeMap<cdk::Amount, cdk01::MintKeyPair> = BTreeMap::default();
+    for (val, keypair) in keys {
+        let uval = val.parse::<u64>().expect("Failed to parse amount");
```

Comment:
This is a workaround for surreal not allowing u64 as key, correct? If so, maybe an explaining comment in this line and line 33 (where amount it to_string`d) would be helpful, since this basically explains why this conversion can't fail.